R/pension_supplement.R

```diff
@@ -0,0 +1,103 @@
+#' 
+#' @author Matthew Katzen
+#' @export
+
+#note: annual = rate multiplied by 364/14 (= 26)
+
+pension_supplement <- function(has_partner = FALSE,
+                               age = 70,
+                               n_dependants = 0,
+                               parenting_payment = FALSE,
+                               Date = NULL,
+                               fy.year = NULL,
+                               age_pension_age_requirement = FALSE,
+                               disability_support_pension = FALSE,
+                               per = 'fortnight',
+                               overseas_absence = FALSE,
+                               seperated_couple = FALSE){
+  
+  if(!(per %in% c('fortnight', 'annual'))){
+    stop("per can only take values `fortnight` or `annual`")
+  }
+  
+  if(any(!has_partner & seperated_couple)) {
+    stop("incompatible values of `has_partner` and `partner_seperated`")
+  }
+  
+  if (is.null(Date)) {
+    if (is.null(fy.year)) {
+      Date <- as.Date("2016/03/01")
+      fy.year <- "2015-16"
+      warning('`Date` and `fy.year` not set, so using `Date` = "2016/03/01" and `fy.year` = "2015-16"')
+    } else {
+      Date <- fy2date(fy.year)
+      warning('`Date` not set. Using date as defined in fy2date()')
+    }
+  } else {
+    if (is.null(fy.year)) {
+      fy.year <- date2fy(Date)
+    } else {
+      warning("`fy.year` and `Date` both used. Ignoring `fy.year`.")
+    }
+  }
+  
+  input <- data.table(do.call(cbind.data.frame, mget(ls()))) #convert arguments to data table
+  
+  
+  input[ ,age_pension_age :=  #increase:https://www.humanservices.gov.au/individuals/services/centrelink/age-pension/eligibility-payment-rates/age-rules
+                              #women increase before 2014 (not implemented): https://www.humanservices.gov.au/sites/default/files/documents/co029-0907.pdf
+    if_else(Date < "2017-07-01",
+            65,
+            if_else(Date < "2019-01-01",
+                    65.5,
+                    if_else(Date < "2021-07-01",
+                            66,
+                            if_else(Date < "2023-07-01",
+                                    66.5,
+                                    67))))]
+
+  input[, eligible := if_else(age_pension_age_requirement,
+                              age >= age_pension_age,
+                              if_else(disability_support_pension,
+                                      !(age < 21 & n_dependants == 0),#ineligble if under 21 and have no kids
+                                      TRUE))]
+
+  input[ ,max_rate_March_2016 :=
+    if_else(has_partner & !seperated_couple,
+            49,
+            65)]
+
+  input[, basic_rate_March_2016 :=
+    if_else(has_partner,
+            18.7,
+            22.70)]
+
+  output <- input[, if_else(eligible,
+                            if_else((!has_partner & parenting_payment) | overseas_absence,
+                                    if_else(per == 'fortnight',
```

Comment:
See #110

Reply:
But since `per` is also a column in input shouldn't it be included in the if_else? Was thinking if there was the case where 
```r
per =  c('fortnight', 'annual')
```